plugin.py

```diff
@@ -101,15 +101,14 @@ def config(self) -> ClientConfig:
                 }
             ],
             settings: {
-                "validate": True,
+                "validate": "on",
                 "packageManager": "npm",
-                "autoFix": True,
-                "autoFixOnSave": True,
                 "options": {},
                 "run": "onType",
                 "nodePath": None,
+                "format": False,
                 "quiet": False,
-                "workspaceFolder": None,
+                "onIgnoredFiles": "off",
                 "codeAction": {
                     "disableRuleComment": {
                         "enable": True,
```

Comment:
Not ideal that those default settings are duplicated here and in the configuration file.

Reply:
I agree, but I don't have a solution for this right now. Until someone has something better, I am ok with this.

Reply:
Sure.

I was also thinking that `LSP-eslint.sublime-settings` could (should) have higher-level settings instead. So instead of passing client options verbatim, it would have a structure like:

```js
{
    enabled: Boolean,
    scopes: Array,
    syntaxes: Array,
    initializationOptions: Object,
    workspaceConfiguration: Object,
    workingDirectory: Object,  // should be support on the project level
}
```

The reason for that is I think it would be better if user could override those individually instead of having to override whole `client` object and make sure that all defaults are provided (that can break easily on updating LSP-eslint to a new version).

Reply:
So it will become something like this:
```
    def config(self) -> ClientConfig:
        settings = sublime.load_settings("LSP-eslint.sublime-settings")

        configuration = {
            "command": [
                'node',
                server_path,
                '--stdio'
            ],
            "languages": settings.get('languages'),
            "initializationOptions": settings.get('initializationOptions'),
            "settings": settings.get('settings')
        }

        return read_client_config('lsp-eslint', configuration)
```

I agree this is much nicer.
(maybe for a different topic, I just realized that we don't really need the `enabled` flag for the Language Handlers, because they are already plugins, and we can always disable plugins with `Package Control: Disable package`...)

Reply:
But because this is a breaking change, we can write a migration, that would transform the old settings file to the new one.
So this:
```
client: {
  a: 1,
  b: 2,
}
```
Should become:
```
{
  a: 1,
  b: 2
}
```
And probably this way of reading settings should be done for all the LSP-* plugins :)

Reply:
> (maybe for a different topic, I just realized that we don't really need the `enabled` flag for the Language Handlers, because they are already plugins, and we can always disable plugins with `Package Control: Disable package`...)

We don't need to expose it in user settings, yes, but we still need to pass it to enable language server. Just checked and doesn't start without it.

Reply:
I've added migration code but I'm not migrating old `settings` object (preserving as `old_settings` only) as that would require too much work to validate that all options have valid value (some of them changed from true/false to "on", "off", "something" for example). It's safer to just use defaults and let user update them if needed.